src/main/java/sx/blah/discord/api/internal/Requests.java

```diff
@@ -56,62 +58,67 @@
 	 */
 	public static final Requests GENERAL_REQUESTS = new Requests(null);
 
+	/**
+	 * The HTTP client requests are made on.
+	 */
+	private final OkHttpClient HTTP;
+
 	/**
 	 * Used to send POST requests.
 	 */
-	public final Request POST;
+	public final DiscordRequest POST;
 	/**
 	 * Used to send GET requests.
 	 */
-	public final Request GET;
+	public final DiscordRequest GET;
 	/**
 	 * Used to send DELETE requests.
 	 */
-	public final Request DELETE;
+	public final DiscordRequest DELETE;
 	/**
 	 * Used to send PATCH requests.
 	 */
-	public final Request PATCH;
+	public final DiscordRequest PATCH;
 	/**
 	 * Used to send PUT requests.
 	 */
-	public final Request PUT;
-
-	/**
-	 * The client used for these requests.
-	 */
-	private final DiscordClientImpl client;
+	public final DiscordRequest PUT;
 
 	public Requests(DiscordClientImpl client) {
-		this.client = client;
+		HttpLoggingInterceptor httpLogger = new HttpLoggingInterceptor(message -> Discord4J.LOGGER.debug(LogMarkers.API, message)).setLevel(HttpLoggingInterceptor.Level.HEADERS);
+		httpLogger.redactHeader("Set-Cookie");
+		httpLogger.redactHeader("Cookie");
+		httpLogger.redactHeader("Authorization");
+		HTTP = new OkHttpClient.Builder().addInterceptor(httpLogger).build();
+
+		POST = new DiscordRequest(RequestMethod.POST, client);
+		GET = new DiscordRequest(RequestMethod.GET, client);
+		DELETE = new DiscordRequest(RequestMethod.DELETE, client);
+		PATCH = new DiscordRequest(RequestMethod.PATCH, client);
+		PUT = new DiscordRequest(RequestMethod.PUT, client);
+	}
 
-		POST = new Request(HttpPost.class, client);
-		GET = new Request(HttpGet.class, client);
-		DELETE = new Request(HttpDelete.class, client);
-		PATCH = new Request(HttpPatch.class, client);
-		PUT = new Request(HttpPut.class, client);
+	/**
+	 * The method types available for a request.
+	 */
+	public enum RequestMethod {
+		POST, GET, DELETE, PATCH, PUT
 	}
 
 	/**
 	 * A specific HTTP method request type.
 	 */
-	public final class Request {
+	public final class DiscordRequest {
```

Comment:
This class can be private

Reply:
I tried changing the access modifier to private and got errors when compiling the jar. I am open to using other access modifiers if applicable.